### PR TITLE
detect/analyzer: add more details for tcp_mss

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6355

Describe changes:
- detect/analyzer: add more details for tcp_mss

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1759

https://github.com/OISF/suricata/pull/10826 with even better names

### CODE REVIEW

**Comment by victorjulien** [comment]
Final request: think this should be 2 commits: 1 for the tojson feature, 1 for the tcp mss support.

File: rust/src/detect/tojson/mod.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::detect::uint::{DetectIntType, DetectUintData, DetectUintMode};
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+
+pub fn detect_uint_to_json<T: DetectIntType>(
```
Comment (jasonish) [inline rust/src/detect/tojson/mod.rs:21]: Does this require a whole new module? Why not just a tojson method on DetectUintMode?

File: rust/src/detect/tojson/mod.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            js.set_uint("mask", du.arg1.into())?;
+            js.set_uint("value", du.arg2.into())?;
+            js.close()?;
+        }
+    }
+    Ok(())
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_detect_u16_to_json(
```
Comment (jasonish) [inline rust/src/detect/tojson/mod.rs:75]: New extern functions should following C naming and namspacing.. So `SC...`.

File: rust/src/detect/tojson/mod.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/detect/tojson/mod.rs:21]: So that uint does not depend on json builder

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20028

File: rust/src/detect/tojson/mod.rs (hunk 2 above)
Comment (jufajardini) [reply rust/src/detect/tojson/mod.rs:75]: Curious: Does this apply to any such functions, that is, functions used for adding new keywords, or other functions that could be needed for particular parsers, should those also follow this?

Comment (jasonish) [reply rust/src/detect/tojson/mod.rs:75]: > Curious: Does this apply to any such functions, that is, functions used for adding new keywords, or other functions that could be needed for particular parsers, should those also follow this?

Essentially any `pub extern "C"` becomes part of our C API, so should follow that format.  Pure Rust function, and even non-pub extern fn's that are used as function pointers can follow Rust convention are they are subject to Rust's visibility rules and namespacing.  `pub extern` will enter the global namespace, and its best we stick to a convention here whether C or Rust.

Comment (catenacyber) [reply rust/src/detect/tojson/mod.rs:75]: Could some tool check this ? Like `cargo fmt` with some config ?

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10841

**Pull request closed without merging**